File: go/kms/DecryptData/DecryptData.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0
// snippet-start:[kms.go.decrypt_data]
package main

// snippet-start:[kms.go.decrypt_data.imports]
import (
    "flag"
    "fmt"

    "github.com/aws/aws-sdk-go/aws/session"
    "github.com/aws/aws-sdk-go/service/kms"
    "github.com/aws/aws-sdk-go/service/kms/kmsiface"
)
// snippet-end:[kms.go.decrypt_data.imports]

// DecodeData decrypts some text that was encrypted with an AWS Key Management Service (AWS KMS) customer master key (CMK).
// Inputs:
//     svc is an AWS KMS service client
//     blob is an array of bytes containing the text to decrypt
// Output:
//     If success, information about the text and nil
//     Otherwise, nil and an error from the call to Decrypt
func DecodeData(svc kmsiface.KMSAPI, blob *[]byte) (*kms.DecryptOutput, error) {
    // snippet-start:[kms.go.decrypt_data.call]
    result, err := svc.Decrypt(&kms.DecryptInput{
        CiphertextBlob: *blob,
    })
    // snippet-end:[kms.go.decrypt_data.call]
    if err != nil {
        return nil, err
    }

    return result, err
}

func main() {
    // snippet-start:[kms.go.decrypt_data.args]
    data := flag.String("d", "", "The encrypted data, as a string")
    flag.Parse()

    if *data == "" {
        fmt.Println("You must supply the encrypted data as a string")
        fmt.Println("-d DATA")
        return
    }
    // snippet-end:[kms.go.decrypt_data.args]

    // snippet-start:[kms.go.decrypt_data.session]
    sess := session.Must(session.NewSessionWithOptions(session.Options{
        SharedConfigState: session.SharedConfigEnable,
    }))

    svc := kms.New(sess)

    blob := []byte(*data)
    // snippet-end:[kms.go.decrypt_data.session]

    result, err := DecodeData(svc, &blob)
    if err != nil {
        fmt.Println("Got error decrypting data: ", err)
        return
    }

    // snippet-start:[kms.go.decrypt_data.display]
    fmt.Println(string(result.Plaintext))
    // snippet-end:[kms.go.decrypt_data.display]
}
// snippet-end:[kms.go.decrypt_data]

File: ruby/example_code/kms/decrypt_data.rb
# frozen_string_literal: true

# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# Purpose:
# decrypt_data.rb demonstrates how to decrypt a string
# using Amazon Key Management Services (AWS KMS) using the AWS SDK for Ruby.

# snippet-start:[kms.ruby.decryptBlob]

require "aws-sdk-kms" # v2: require 'aws-sdk'

# Decrypted blob

blob = "01020200785d68faeec386af1057904926253051eb2919d3c16078badf65b808b26dd057c101747cadf3593596e093d4ffbf22434a6d00000068306606092a864886f70d010706a0593057020100305206092a864886f70d010701301e060960864801650304012e3011040c9d629e573683972cdb7d94b30201108025b20b060591b02ca0deb0fbdfc2f86c8bfcb265947739851ad56f3adce91eba87c59691a9a1"
blob_packed = [blob].pack("H*")

client = Aws::KMS::Client.new(region: "us-west-2")

resp = client.decrypt({
                        ciphertext_blob: blob_packed
                      })

puts "Raw text: "
puts resp.plaintext
# snippet-end:[kms.ruby.decryptBlob]
